src/os/filestore/FileStore.cc

```diff
@@ -1873,10 +1898,17 @@ void FileStore::_do_op(OpSequencer *osr, ThreadPool::TPHandle &handle)
 
   osr->apply_lock.Lock();
   Op *o = osr->peek_queue();
-  apply_manager.op_apply_start(o->op);
+
+  if (!((g_conf->filestore_do_fast_sync) && (m_filestore_journal_writeahead))) {
+    apply_manager.op_apply_start(o->op);
```

Comment:
Once you've added a fast_sync member to ApplyManager, it would be better if here (and other similar places) called op_apply_start unconditionally, but the implementation was a noop based on ApplyManager::fast_sync.  My reasoning is that a future reader of the code looking at the ApplyManager implementation would reasonably assume that op_apply_start is called whenever an op is about to be applied.